holidays/calendars.py

```diff
@@ -942,3 +945,1420 @@ def ok_phansa_date(self, year: int) -> Optional[date]:
         else:
             delta_days = 309
         return start_date + td(days=delta_days)
+
+
+class _IslamicHolidays:
```

Comment:
We use *Holidays for holiday_groups normally. I guess these entities are more lie calendars. I'd use something similar, e.g., IslamicLunar